idea-plugin/src/com/reason/ocaml/OCamlFormatProcess.java

```diff
@@ -0,0 +1,46 @@
+package com.reason.ocaml;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.ScriptRunnerUtil;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.reason.Log;
+import com.reason.ide.settings.ORSettings;
+import org.jetbrains.annotations.NotNull;
+
+public class OCamlFormatProcess {
+
+    private static final Log LOG = Log.create("ocamlformat.process");
+
+    private final Project m_project;
+
+    public static OCamlFormatProcess getInstance(@NotNull Project project) {
+        return ServiceManager.getService(project, OCamlFormatProcess.class);
+    }
+
+    public OCamlFormatProcess(Project project) {
+        m_project = project;
+    }
+
+    public String format(@NotNull PsiFile psiFile) {
+        String ocamlformatExecutable = ORSettings.getInstance(m_project).getOcamlformatExecutable();
```

Comment:
I think that it should use be done like the dune process (must work with wsl/cygwin): it should use the opam env.
also, if nothing is set in the settings, it should try to use the installed ocamlformat from opam.